docs/examples/03_building_custom_image.ipynb

```diff
@@ -0,0 +1,253 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Building custom image for function\n",
+    "\n",
+    "In this tutorial we will describe how to build custom docker image for function.\n",
+    "\n",
+    "In this tutorial we will be following 3 steps to deploy our function with custom docker image:\n",
+    "1. implement function template \n",
+    "2. define dockerfile, build it and push to registry\n",
+    "3. upload"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "First we will implement function entrypoint by following template. All functions with custom docker images must follow same template structure. \n",
+    "\n",
+    "We need to create class `Runner` and implement `run` method that will be called during invocation of the function and results of the run method will be returned as result of the function.\n",
+    "\n",
+    "Let's create [runner.py](./custom_function/runner.py) file with following content\n",
+    "```python\n",
+    "class Runner:\n",
+    "    def run(self, arguments: dict) -> dict:\n",
+    "        # this is just an example\n",
+    "        # your function can call for other modules, function, etc.\n",
+    "        return {\n",
+    "            **arguments,\n",
+    "            **{\n",
+    "                \"answer\": 42\n",
+    "            }\n",
+    "        }\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As a next step let's define and build our custom docker image.\n",
+    "\n",
+    "Dockerfile will be extending base serverless node image and adding required packages and structure to it. \n",
+    "\n",
+    "In our simple case it will look something like this\n",
+    "\n",
+    "```Dockerfile\n",
+    "FROM icr.io/quantum-public/quantum-serverless-ray-node:0.9.0-py39\n",
```

Comment:
Bump docker image tag to 0.10.1-py310?

Reply:
and add to update-component-versions action?